一、RTS3交易测试脚本，收到成交回报推送时，ParseFromString解析报错，有时报错，有时包解析正常，错误如下：
Unhandled exception in thread started by <function ReceiveMsg at 0x0000000002674
3C8>
Traceback (most recent call last):
  File "test_stock_order(buy).py", line 78, in ReceiveMsg
    elif func_id==0x0007:
  File "build\bdist.win-amd64\egg\google\protobuf\message.py", line 179, in Pars
eFromString
  File "build\bdist.win-amd64\egg\google\protobuf\internal\python_message.py", l
ine 755, in MergeFromString
  File "build\bdist.win-amd64\egg\google\protobuf\internal\python_message.py", l
ine 782, in InternalParse
  File "build\bdist.win-amd64\egg\google\protobuf\internal\decoder.py", line 410
, in DecodeField
UnicodeDecodeError: 'utf8' codec can't decode byte 0x86 in position 8: invalid s
tart byte

二、Beautiful Soup解析网页时，某些数据由js生成，如链接：
http://www.sse.com.cn/disclosure/diclosure/etf/index.shtml?type=1&etfClass=1
下列表格中td标签内无值，如何获取td标签内的值？
<table class="tablestyle">
    <tr>
        <td class="th_l2" width="50%">最小申购、赎回单位的预估现金部分(单位:元)</td>
        <td id="estimatedCashComponent" width="50%"></td>
    </tr>
    <tr>
        <td class="th_l2" >现金替代比例上限</td>
        <td id="maxCashRatio"></td>
    </tr>
    <tr>
        <td class="th_l2" >申购上限</td>
        <td id="creationLimit"></td>
    </tr>
    <tr>
        <td class="th_l2" >赎回上限</td>
        <td id="redemptionLimit"></td>
    </tr>
    <tr>
        <td class="th_l2" >是否需要公布IOPV</td>
        <td id="publishIopv"></td>
    </tr>
    <tr>
        <td class="th_l2" >最小申购、赎回单位(单位:份)</td>
        <td id="creationRedemptionUnit"></td>
    </tr>
    <tr>
        <td class="th_l2" >申购赎回的允许情况</td>
        <td id="creationRedemption"></td>
    </tr>
</table>

解决方法：
安装PyQt4和spynner，获取动态网页，再用BeautifulSoup解析，见链接http://www.comingcode.com/?p=275

三、使用spynner模块下载网页时，报错：
In [6]: import spynner

In [7]: browser = spynner.Browser()

In [8]: browser.load("http://www.wordreference.com")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
C:\Python27\lib\site-packages\spynner\browser.pyc in createRequest(manager, oper
ation, request, data)
   1672         )
   1673         manager.cookieJar().setAllCookies(cookies)
-> 1674         url = six.u(toString(request.url()))
   1675         operation_name = self._operation_names.get(
   1676             operation, str(operation)).upper()

C:\Python27\lib\site-packages\six-1.9.0-py2.7.egg\six.pyc in u(s)
    619     # Workaround for standalone backslash
    620     def u(s):
--> 621         return unicode(s.replace(r'\\', r'\\\\'), "unicode_escape")
    622     unichr = unichr
    623     int2byte = chr

TypeError: decoding Unicode is not supported

解决方法：
C:\Python27\lib\site-packages\six-1.9.0-py2.7.egg\six.py 目录下该.py文件中，原函数：
def u(s):
    return unicode(s.replace(r'\\', r'\\\\'), "unicode_escape")
    
改为：
def u(s):
    try:
        return unicode(s.replace(r'\\', r'\\\\'), "unicode_escape")
    except TypeError as e:
        if "decoding Unicode is not supported" in str(e):
            return unicode(s.replace(r'\\', r'\\\\'))
            
再运行代码：
In [1]: import spynner

In [2]: browser = spynner.Browser()

In [3]: browser.load("http://www.wordreference.com")
Out[3]: True


四、运行下列代码报错
import spynner
import pyquery
import time
from bs4 import BeautifulSoup

browser = spynner.Browser()
browser.create_webview()
browser.set_html_parser(pyquery.PyQuery)
browser.load("http://www.sse.com.cn/disclosure/diclosure/etf/index.shtml?type=1&etfClass=1", 5)

try:
	browser.wait_load(5)
except:
	pass

string = browser.html 
browser.close() 
soup = BeautifulSoup(string)
time.sleep(1)

trading_day = soup.find(id="tradingDay")
print trading_day

结果：
TypeError: _on_webview_destroyed() takes exactly 2 arguments (1 given)
<td class="content_warning " id="tradingDay" width="50%">2015-07-23</td>
[Finished in 5.2s]

解决方法：
windows64bit机器上，把原先的PyQt4-4.11.4-cp27-none-win_amd64卸载，重新安装PyQt4-4.10.1-gpl-Py2.7-Qt4.8.4-x64.exe
spynner卸载，重新安装spynner-2.5，运行脚本，不报错


